pac4j-core/src/main/java/org/pac4j/core/util/CommonHelper.java

```diff
@@ -83,7 +83,7 @@ public static boolean areEquals(final String s1, final String s2) {
     public static boolean areEqualsIgnoreCaseAndTrim(final String s1, final String s2) {
         if (s1 == null && s2 == null) {
             return true;
-        } else if (s1 != null && s2 !=  null) {
+        } else if (s1 != null && s2 != null) {
             return s1.trim().equalsIgnoreCase(s2.trim());
         } else {
```

Comment:
In fact, this check is necessary. Let's take an example: s1 = "me", s2 = null;

s1 == null && s2 == null -> false -> else case -> s2.trim() generates a NullPointerException